File: 0011-container-with-most-water/0011-container-with-most-water.rb
def max_area(height)
  left = 0
  right = height.size - 1
  result = 0

  while left < right
    result = [result, [height[left], height[right]].min * (right - left)].max
    if height[left] < height[right]
      left += 1
    else
      right -= 1
    end
  end
  result
end

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = result = 0 
        right = len(heights) - 1
        

        while left < right:
            result = max(result, min(heights[left], heights[right]) * (right - left))
            if heights[left] < heights[right]:
                left += 1
            elif heights[right] <= heights[left]:
                right -= 1
            
        return result
